pkg/migrator/core_test.go

```diff
@@ -235,3 +240,68 @@ func TestMigrateListClusterScoped(t *testing.T) {
 		}
 	}
 }
+
+type fakeProgress struct{}
+
+func (f *fakeProgress) load() (string, error) {
+	return "", nil
+}
+
+func (f *fakeProgress) save(string) error {
+	return nil
+}
+
+func TestMetrics(t *testing.T) {
+	metrics.Metrics.Reset()
+	// fake client doesn't support pagination, so we can't test complex behavior.
+	nodeList := newNodeList(100)
+	client := fake.NewSimpleDynamicClient(scheme.Scheme, toUnstructuredListOrDie(nodeList))
+	migrator := NewMigrator(v1.SchemeGroupVersion.WithResource("nodes"), client, &fakeProgress{})
+	migrator.Run()
+	expectCounterCount(t,
+		"storage_migrator_core_migrator_migrated_objects",
+		map[string]string{
+			"resource": "/v1, Resource=nodes",
+		},
+		100,
+	)
+
+}
+
+func labelsMatch(metric *ptype.Metric, labelFilter map[string]string) bool {
```

Comment:
If we have labelFilters with keys __*not*__ present in the metric labels, shouldn't we also return false?

Reply:
Fixed. These functions are copied from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/metrics/testutil_test.go#L26. We probably should fix it there, too.